File: zadanie1-2.c
// Autor: Daniel Sawa
// Ciel programu:
//
//
#include<stdio.h>

int main() {
    char ch;

    int vstup, i = 0;
    int sucet = 0;
    int n;
    int *pole_vstupu;

    



    while(2 == scanf("%d%c", &vstup, &ch)){
        if(i == 0){
            n = vstup;
            pole_vstupu = (int*)malloc(n * sizeof(int));
        }
        else{
            int j = i-2;
            while (j >= 0 && pole_vstupu[j] > vstup) {
                pole_vstupu[j + 1] = pole_vstupu[j];
                j = j - 1;
            }
            pole_vstupu[j + 1] = vstup;
        }
        i++;
        if(ch == '\n'){    
            i = 0;
            int pocitadlo = 0;
            int pocet_kamionov = 0;
            int od_zadu = 0;
            
            for (int j = n-1; j > 0; j--)
            {
                if (pole_vstupu[j] >= 200)
                {
                    pocet_kamionov++;
                    pocitadlo++;
                }
                else
                {
                    if(j <= od_zadu){
                        break;
                    }
                    pocet_kamionov++;

                    if (pole_vstupu[j] + pole_vstupu[od_zadu] > 300)
                    {
                        pocitadlo++;
                        continue;
                    }
                    pocitadlo += 2;
                    od_zadu++;   
                }
            }
            if (n - pocitadlo == 1)
            {
                pocet_kamionov++;
            }
            
            printf("%d \n",pocet_kamionov);
        }
        else if(ch == '$'){
            break;
        }
    }
}

File: zadanie1-2.py
#vstup = [ 120 ,140, 150, 200 ,201, 250 ,150 ,150 ,101, 110,160 ,175,130 ,185, 190, 300]
vstup = [ 120 ,140, 150, 200 ,201, 250 ,150 ,150 ,160 ,175,130 ,185, 190, 300]


# 14 120 140 150 200 201 250 150 150 160 175 130 185 190 300

vstup.sort()
vstup.reverse()

print(vstup)

pocet_kamionov = 0
od_ktoreho = 0
od_zadu = 0

for idx,i in enumerate(vstup):
    if i >= 200:
        od_ktoreho += 1
    else:
        if idx >= len(vstup)-od_zadu-1:
            break
                
        print(len(vstup)-od_zadu-1)
        pocet_kamionov += 1

        if i + vstup[len(vstup)-od_zadu-1] > 300:
            continue
        
        
        print(vstup[len(vstup)-od_zadu-1] + i,idx,len(vstup)-od_zadu-1)
        od_zadu +=1



pocet_kamionov += od_ktoreho


print(pocet_kamionov)